docs/TestRunner.md

```diff
@@ -0,0 +1,131 @@
+# Test Parser
+
+## Overview 
+
+Parse a [yml file](https://docs.google.com/document/d/17GOv1yVAKY4vmOd1Vhg_IitpyCMiX-e_b09eufNysYI/edit#heading=h.9dwsxj6quakr) into a test object.
+
+## How to use it
+
+Create a TestParser using the path of the yml file.
+
+```
+const TestParser = require('skill-testing-ml').TestParser;
+const parser = new TestParser("en-US/edgar.yml");
+
+```
+
+Call the parse method, if the file is valid an object will be created, if not an exception will be through with the error.
+
+```
+parser.parse();
+```
+
+Error example:
+```
+Error - YAMLException:
+end of the stream or a document separator is expected at line 17, column 14:
+    configuration:
+
+```
+
+
+# Test Runner
+
+## Overview
+Test Runner allows to execute tests using [SkillTestingMarkupLanguage](https://docs.google.com/document/d/17GOv1yVAKY4vmOd1Vhg_IitpyCMiX-e_b09eufNysYI/edit#heading=h.9dwsxj6quakr).
+
+## How to used it
+
+Create a TestRunner instance with the required configuration.
+
+```
+const TestRunner = require('skill-testing-ml').TestRunner;
+
+const runner = new TestRunner({
+    invoker: "virtualDeviceInvoker",
+});
+
+```
+
+Call run method using the file path and a context object, context object will be sent back as part of the event.
+```
+await runner.run("fact-skill-tests.yml", context);
+
+```
+
+
+TestRunner will emit events before and after each testcase of the file, use subsribe method to add callbacks and listen for the events. "message" event will be fired before the execution and "result" after that. 
```

Comment:
"will emit events before and after each test interaction is run in the file"

There is also a typo on subscribe "subsribe"

"message" event will be fired before each interaction, and result will be sent back after each reply.

I also just reviewed the code for doing the messages and results - it is not correct. It needs to be handing back results as each interaction happens. I am going to create a separate issue for this.